src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/UpToDateCheckImplicitConfiguredInput.cs

```diff
@@ -488,6 +445,50 @@ ImmutableArray<string> UpdateResolvedAnalyzerReferencePaths()
                 return ResolvedAnalyzerReferencePaths;
             }
 
+            (ImmutableArray<string> ResolvedCompilationReferencePaths, ImmutableArray<string> CopyReferenceInputs) UpdateResolvedCompilationReferences()
+            {
+                if (jointRuleUpdate.ProjectChanges.TryGetValue(ResolvedCompilationReference.SchemaName, out IProjectChangeDescription? change) && change.Difference.AnyChanges)
```

Comment:
nit: Also consider flipping to be a guard clause instead of if-else.

Reply:
See above.